### DISCUSSION TITLE
Report emulator compatibility issues here

### DISCUSSION THREAD

**Discussion by garungorp** [garungorp/MouseInjectorDolphinDuck#3] [General]
Please post any issues pertaining to emulator compatibility here.

**Issue Examples**
- A game that works with one supported emulator but not another
- An update to an emulator breaks compatibility

**Comment by RoseVelika** [comment]
Signed back into my account after years just to post about this because it's making me lose it, I downloaded an update to the injector, copied the stuff over and not only does the new update not work with King's Field (II) going back to the old one doesn't work either I just keep getting this error no matter how many times I delete duckstation or the injector and remove their settings.
![image](https://user-images.githubusercontent.com/13590521/223214115-8bb61f86-65c1-4afb-b077-9485db48c50c.png)

**Reply by RoseVelika** [reply]
Ok the issue seems to be even bigger than I realized, the injector won't hook in ANY game regardless of emulator and I have no idea how to fix this.

**Reply by garungorp** [reply]
Did you try clicking the '**Restore Defaults**' button in the DuckStation settings? The settings are kept in the Documents folder and will persist even when deleting the DuckStation folder. This is a problem a few people have had and this has solved it as some setting changes the memory and makes the injector unable to find the correct memory region. Also the injector doesn't need to be copied anywhere as it will work from whichever directory is extracted to.

Let me know if this helps.

**Reply by RoseVelika** [reply]
I made sure to do that the very first time and it still didn't fix anything, for some reason it just isn't working and I'm not entirely sure why.

**Reply by RoseVelika** [reply]
![image](https://user-images.githubusercontent.com/13590521/223225021-b0166d47-11a5-46ee-8eb1-7d68c78b7e74.png)
Here's me showing it with the game in the background, it does this even when it's in-game

**Reply by garungorp** [reply]
I thought it may have been due to you using the undub version but it seems to work with that one as well so if it doesn't work with any game or emulator and it was working before then it seems to be a system wide issue.

It could possibly be due to anti-virus running as it does affect the memory of other programs but other than that I'm not sure as I can't reproduce the problem. Some emulators are finicky when hooking like BSNES as it changes the memory structure often to where it takes a specific set of steps to hook but DuckStation with default settings, PCSX2 with a compatible BIOS, and Dolphin all work pretty reliably and I haven't come across an issue like this.

Sorry I can't help you at the moment.

**Reply by KarlGhartler** [reply]
Unlikely to fix it for you if its happening in all emulators, but I fixed my Duckstation not hooking by creating a separate portable install with default settings

**Reply by RoseVelika** [reply]
Yeah none of the new updates are working for me either, tried on all emus again...
I guess it's just a system thing but I have no idea what it could be...

**Reply by Bosniankid** [reply]
This worked for me on PCSX2 What I did was to disabled "fast boot" on the emulator and let it to display the whole bios booting, then when it shows the "PlayStation 2" logo just in that moment I activate the injector with 4 key number, when the PS2 logo disappears the injector will finally detect the game.

**Reply by shevoobrazilian** [reply]
> Yeah none of the new updates are working for me either, tried on all emus again... I guess it's just a system thing but I have no idea what it could be...

I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Reply by shevoobrazilian** [reply]
> Signed back into my account after years just to post about this because it's making me lose it, I downloaded an update to the injector, copied the stuff over and not only does the new update not work with King's Field (II) going back to the old one doesn't work either I just keep getting this error no matter how many times I delete duckstation or the injector and remove their settings. ![image](https://user-images.githubusercontent.com/13590521/223214115-8bb61f86-65c1-4afb-b077-9485db48c50c.png)

I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

só uma dica: se vc for alterar o controle do emulador, lembre se de sempre apertar no botão ''aplicar perfil'', se estiver desmarcado após vc configurar, já que o emulador só altera se vc apertar esse botão.

**Comment by IDrinkLava** [comment]
I'm trying to get Black working in PCSX2 but it's just not recognizing it. I have the exact nightly build, I've tried both the SCPH-90001 and SCPH-70012 BIOS, but it just keeps telling me "no game loaded".

**Reply by garungorp** [reply]
Some settings will prevent the emulator from hooking so I would first recommend clicking the 'Restore Defaults' button in the settings and then restarting the emulator and the injector.

**Reply by tinotrev** [reply]
> I'm trying to get Black working in PCSX2 but it's just not recognizing it. I have the exact nightly build, I've tried both the SCPH-90001 and SCPH-70012 BIOS, but it just keeps telling me "no game loaded".

The same thing is happening to me. I haven't been able to fix it yet.

**Comment by AkitoFire** [comment]
I have the exact same issue as @RoseVelika, I tried Duckstation and PCSX2 and the Mouse Injector correctly detects the respective emulators but unfortunately none of the games I've tried (KF1-4, Shadow Tower) work. I only get "No game loaded". Both Duckstation and PCSX2 were fresh installs with the newest versions. 'Restore Defaults' in Duckstation didn't work either.

**Reply by garungorp** [reply]
I'm not sure of the exact problem as I can't reproduce it unfortunately. I've only tested it on Windows 10 so if you're using another version of Windows then that might be an issue. 

A few things you could try:

- Run as administrator
- Disable anti-virus or add the program to the exclusion list
- Run in compatibility mode if you're on Windows 11

Sorry I can't help more but let me know if you try the steps above and if they work for you.

**Reply by AkitoFire** [reply]
Unfortunately, none of these things worked, I'm on W10 too. I also closed pretty much all other applications, in case one of them caused any problems. I also realized that my PCSX2 used an unsupported BIOS, but fixing that also didn't work. Appreciate your help though, based on the footage I saw on your Youtube channel this is definitely the way I'd want to experience King's Field (and many other already supported games)! So thank you for your work, if I eventually find a fix I will post something here.

**Reply by illtron99** [reply]
Having this same exact issue myself over a year later

**Reply by ElTioRata** [reply]
I am currently having this issue with PCSX2. Can't get it to hook up to any game.

**Reply by TonyTonsils** [reply]
This is happening to me with Red Dead Revolver. Last night it was working fine but I had to reinstall PCSX2 to stop it from thinking the spacebar was still the pause hotkey. Today, with a fresh install and that spacebar issue solved, NOW the mouse injector decides to stop working.

**Reply by shevoobrazilian** [reply]
I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Reply by shevoobrazilian** [reply]
> I am currently having this issue with PCSX2. Can't get it to hook up to any game.

I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Reply by shevoobrazilian** [reply]
> Having this same exact issue myself over a year later

I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Comment by Ryoma123456789** [comment]
I'm playing all of Coded Arms and I've noticed points where the mouse injector stops working: the first time it happened to me I don't remember what level, and the second time was against the first boss of the game.

**Comment by lightindarkness435** [comment]
hello i was wondering if you could help me i read a few of the posts i removed anti virus programs. when i run the mouse injector it says game not loaded into pcsx2

**Comment by DfactorLongplays** [comment]
Hiya, I'm having issues with patching 007: The World is Not Enough.
The original MD5 hash is [7B1E0D16D3FB734A15F8D861D8F2EE06](http://redump.org/disc/2387/).

After applying the xdelta, the hash is D57FCFA35B58A8A3DD6836D87A2BB6BA.
In Beetle PSX HW the game gets stuck on loading screen for the first mission. In DuckStation the main menu is immediately corrupted.

Used Delta Patcher 3.0.1 and older versions to no avail.
Unpatched version runs fine.

**Reply by garungorp** [reply]
I'm getting the same hash sums for the original and the patched. Also I'm getting the same results with Beetle PSX HW where it gets stuck during loading. But I cannot reproduce the menu corruption in DuckStation after patching using Delta Patcher. For DuckStation it could possibly be due to some settings so I would recommend restoring the default settings and perhaps try with a different BIOS.

I pretty much exclusively use DuckStation so I was unaware of the issue with Beetle. I will most likely have to adjust the patch to get it to work with Beetle so I'll try to look at that here soon.

**Reply by DfactorLongplays** [reply]
Thank you for the quick response!

I assumed it was a case of a possibly bad xdelta, but I was wrong. The latest version of DuckStation does not render the game properly in Vulkan, while it does work as intended in DX11 and DX12. So that compatibility issue is solely on DuckStation.

As for Beetle PSX, would love a patch if it's within the realm of possibility.

Edit. DuckStation's latest preview build 0.1-5683-g105c1363 seems to have broken Vulkan support and this affects other games too.

**Reply by ldrun06** [reply]
As of this writing, Beetle PSX and Beetle PSX HW are still stuck on the loading screen when I use RetroArch and the patched TWINE game. Latest version of those cores, latest version of Garungorp's mouse injector.

But, when I used PCSX ReARMed, and hit F11 for 'Grab Mouse', the injector was finally recognizing the game. PCSX ReARMed plays this game fine (at least, the first level).

This [https://github.com/vl4sty/MouseInjectorDuck](fork) claims to fix the capturing (hitting '4' at the right time and the game still not being recognized) issue with DuckStation and PCSX2.

**Comment by secondsabre** [comment]
Any luck getting this utility to work in Duckstation versions after 0.1-5943? Been slowly crawling through version iterations, but 5943 seems to be the only one that hooks properly.  I've reset my settings (multiple times) and tried portable installs, but no dice so far.

**Reply by Clippy95** [reply]
https://github.com/vl4sty/MouseInjectorDolphinDuck

**Comment by RAINZFOXX** [comment]
Does anyone know what the last nightly release build of PCSX2 was that still worked with the injector?

**Reply by Clippy95** [reply]
Once again use https://github.com/vl4sty/MouseInjectorDolphinDuck .

**Reply by RAINZFOXX** [reply]
I tried that, it didn't work for me. It won't hook Red Dead Revolver.

**Reply by Clippy95** [reply]
> I tried that, it didn't work for me. It won't hook Red Dead Revolver.

RDR has issues of it's own, I assume you launched PCSX2 then the tool then the game?

**Reply by RAINZFOXX** [reply]
> RDR has issues of it's own, I assume you launched PCSX2 then the tool then the game?

Correct, it wouldn't hook.

**Reply by vl4sty** [reply]
I tried Red Dead Revolver now, and it hooks fine. You must make sure the serial number of the game is SLUS-20500, the US NTSC version. PAL and NTSC-J will not work/hook, the hacks would have to be ported.

**Reply by RAINZFOXX** [reply]
> I tried Red Dead Revolver now, and it hooks fine. You must make sure the serial number of the game is SLUS-20500, the US NTSC version. PAL and NTSC-J will not work/hook, the hacks would have to be ported.

I tried a different BIOS and finally got it to hook.

**Reply by ArhumMK** [reply]
Trying Red Dead too, I noticed Dead eye doesn't work when the hack is active. I can't seem to register the targets with the mouse. I have to disable mouse injector and use the mapped right stick buttons in order to mark targets. Is this a known issue? I don't see it marked so maybe i have something wrong here.

**Reply by Clippy95** [reply]
> Trying Red Dead too, I noticed Dead eye doesn't work when the hack is active. I can't seem to register the targets with the mouse. I have to disable mouse injector and use the mapped right stick buttons in order to mark targets. Is this a known issue? I don't see it marked so maybe i have something wrong here.

Known issue

**Reply by RAINZFOXX** [reply]
> > Trying Red Dead too, I noticed Dead eye doesn't work when the hack is active. I can't seem to register the targets with the mouse. I have to disable mouse injector and use the mapped right stick buttons in order to mark targets. Is this a known issue? I don't see it marked so maybe i have something wrong here.
> 
> Known issue

While we're still on the topic of Red Dead Revolver. Any idea what causes these artifacts? Happens on all the renderers. Switching to Software Mode then switching back fixes it but always seems to happen after very cutscene.
![Red Dead Revolver_SLUS-20500_20241112022512](https://github.com/user-attachments/assets/06332e6c-9bff-4a57-9e52-10ce8cab1a2f)

**Reply by Clippy95** [reply]
Are you using AMD?

**Reply by RAINZFOXX** [reply]
> Are you using AMD?

No, NVIDIA.

**Reply by Clippy95** [reply]
Most likely a regression, does the game go randomly black for you when playing? I think I was able to fix deadeye not working in Mouse Injector but I don't wanna push it to any repo if my 'fix' is causing it.
EDIT: Seems like my fix is the culprit eh

**Reply by RAINZFOXX** [reply]
I was able to fix it by disabling Safe Settings in Renderer Fixes but yeah I am getting constant black screens :(

**Reply by Dirt-Cheap-1-2** [reply]
> I was able to fix it by disabling Safe Settings in Renderer Fixes but yeah I am getting constant black screens :(

Were you ever able to fix this issue?

**Reply by RAINZFOXX** [reply]
No, I ended up giving up on it.

**Reply by shevoobrazilian** [reply]
> > I was able to fix it by disabling Safe Settings in Renderer Fixes but yeah I am getting constant black screens :(
> 
> Were you ever able to fix this issue?

I GOT IT!!!!!

I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Comment by ikozawe** [comment]
Can someone help ?
I have problem on Darkwatch it wont hook I read everything from readme and nothing helped if someone have solution for it please help

**Reply by Clippy95** [reply]
I assume you're using the **required** patch?

**Reply by wejs123** [reply]
Same issue. Fresh PCSX2 nightly install (tried both latest and one from Jan '23 when the [video](https://www.youtube.com/watch?v=SRJNyIWniU4) was posted), checked all the required options, copied .pnach file (it was loaded) and started both programs in correct order in administrator mode and yet mouse injector says "No game loaded". I don't know if this affects anything but the serial listed in the readme is wrong (should be SLUS-21042).

**Comment by Racket2850** [comment]
Hi, the mouse injector is excellent despite it struggling from a glaring compatibility issue, which renders the injector unable to detect any supported game while it's being loaded. It's a type of issue that goes same for both Duckstation and PCSX2, the injector appears to hook whenever it feels like it, regardless of whether it's being run as admin, set in compatibility mode, being run along with a cheat/patch file and/or being excluded from an anti virus service - It does work per se, but it's as though I have to press '4' at a very specific time during the game's bios boot and all versions of said bios was tried. 

As for the emulator versions, the version of Duckstation you suggested for the injector is no longer available on it's releases page from what I can gather. Other than that, the injector works swimmingly. Thanks.

**Reply by vl4sty** [reply]
Hey, use my fork or Lumia's fork. I have fixed the hooking for these emulators. [https://github.com/vl4sty/MouseInjectorDolphinDuck/](https://github.com/vl4sty/MouseInjectorDolphinDuck/)

**Comment by shevoobrazilian** [comment]
> Please post any issues pertaining to emulator compatibility here.
> 
> **Issue Examples**
> 
> * A game that works with one supported emulator but not another
> * An update to an emulator breaks compatibility

I GOT IT!!!!!
I downloaded the stable version 2.2.0 of pcsx2, after that I left the injector mouse open and opened black (USA) as if it were the first time (without save state), I noticed that it recognized black, but I couldn't recover the save state that I had played on the controller because the emulator was in another version, but that's okay!!.

Now I keep the stable version 2.2.0 just for black lol, the night version has the rest of the games;

I'm using BLACK(USA) SLUS-21376 from vimnslair site. https://vimm.net/vault/7970

My bios is SCPH-BIOS 70012. I downloaded it from the ps2 bios site. But I'll leave the link here where I uploaded it too. https://archive.org/details/scph-70012-bios-v-12-usa-200-ntsc

**Comment by jimmypatterson** [comment]
@garungorp Is there anyway we can donate for you to do make the psx duck station mouse injector work for Medal of Honor psx 1999? It works for the first level only but not the rest. :( please help,

**Reply by vl4sty** [reply]
Check my fork. I added basic support.

**Comment by redmaca** [comment]
Return to Castle Wolfenstein: Operation Resurrection (USA) has the Auto Center View issue. It happened when I move forward and  backward. Left and right works fine. How to fix that?
